Project 7 - Bankist/js/script.js

```diff
@@ -0,0 +1,515 @@
+    datalistReceiver.appendChild(optionEl);
+  });
+};
+
+const renderLoginUserList = () => {
+  switchUsersList.innerHTML = "";
+  const userList = allUserExceptCurrentLoggedInUser();
+  userList.forEach((user) => {
+    const div = document.createElement("div");
+    div.classList.add("user__list", "user-avatar-container");
+    div.setAttribute("data-user_id", user.account_id);
+
+    const imgEl = document.createElement("img");
+    imgEl.setAttribute("src", `images/${user.avatar}`);
+    imgEl.setAttribute("id", "current-user-avatar");
+    imgEl.alt = user.name;
+    imgEl.classList.add("user__list-avatar");
+
+    const pEl = document.createElement("p");
+    pEl.classList.add("user__name");
+    pEl.textContent = user.name.split(" ")[0];
+
+    div.appendChild(imgEl);
+    div.appendChild(pEl);
+    switchUsersList.appendChild(div);
+  });
+};
+
+const displayTotalDeposit = () => {
+  const totalDeposit = loggedInUser.transactions
+    .map((transaction) => transaction.transaction_amount)
+    .filter((amount) => amount > 0)
+    .reduce((acc, cur) => acc + cur, 0);
+  labelTotalDeposit.textContent = formatCurrency(
+    loggedInUser.currency,
+    loggedInUser.language,
+    totalDeposit
+  );
+};
+
+const displayTotalWithdrawal = () => {
+  const totalWithdrawal = loggedInUser.transactions
+    .map((transaction) => transaction.transaction_amount)
+    .filter((amount) => amount < 0)
+    .reduce((acc, cur) => acc + cur, 0);
+
+  labelTotalWithdrawn.textContent = formatCurrency(
+    loggedInUser.currency,
+    loggedInUser.language,
+    Math.abs(totalWithdrawal)
+  );
+};
+
+const displayLoanAmount = () => {
+  if (!loggedInUser.loans) {
+    labelLoanAmount.textContent = formatCurrency(
+      loggedInUser.currency,
+      loggedInUser.language
+    );
+    return;
+  }
+  const totalLoan = loggedInUser.loans
+    .map((loan) => loan.principal)
+    .reduce((acc, cur) => acc + cur, 0);
+
+  labelLoanAmount.textContent = formatCurrency(
+    loggedInUser.currency,
+    loggedInUser.language,
+    totalLoan
+  );
+};
+
+const displayStatusMessage = (message, status) => {
+  labelStatusMessage.textContent = message;
+  labelStatusMessage.classList.remove("hidden");
+  labelStatusMessage.classList.add(status);
+  setTimeout(() => {
+    labelStatusMessage.classList.add("hidden");
+  }, 2000);
+};
+
+const displayAvatar = (altText, src) => {
+  imgLoggedInUserAvatar.src = `images/${src}`;
+  imgLoggedInUserAvatar.alt = altText;
+};
+
+const displayWelcomeMessage = (username) => {
+  const now = new Date();
+  const hour = now.getHours();
+  let greeting = "Good Night";
+  if (hour >= 6 && hour <= 10) {
+    greeting = "Good Morning";
+  } else if (hour >= 11 && hour <= 14) {
+    greeting = "Good Day";
+  } else if (hour >= 15 && hour <= 18) {
+    greeting = "Good Afternoon";
+  } else if (hour >= 19 && hour <= 22) {
+    greeting = "Good Evening";
+  }
+  labelWelcome.textContent = `${greeting}, ${username}!`;
+};
```

Comment:
@sknepali You could consider moving the `greetingTimes` array outside of the function. This way, the array isn’t recreated every time the function is called. Here’s how you could do it:
```js
// Define this in different file to manage code effectively
const greetingTimes = [
  { startHour: 6, endHour: 10, greetingMessage: "Good Morning" },
  { startHour: 11, endHour: 14, greetingMessage: "Good Day" },
  { startHour: 15, endHour: 18, greetingMessage: "Good Afternoon" },
  { startHour: 19, endHour: 22, greetingMessage: "Good Evening" },
  { startHour: 23, endHour: 5, greetingMessage: "Good Night" }
];

const displayWelcomeMessage = (username) => {
  // Get the current date and hour
  const currentDate = new Date();
  const currentHour = currentDate.getHours();

  // Find the appropriate greeting based on the current hour
  const currentGreeting = greetingTimes.find(({ startHour, endHour }) =>
    (startHour <= endHour) ? (currentHour >= startHour && currentHour <= endHour) : (currentHour >= startHour || currentHour <= endHour)
  ) || { greetingMessage: "Hello" };

  // Display the greeting message to the user
  labelWelcome.textContent = `${currentGreeting.greetingMessage}, ${username}!`;
};
```
This version of the code should be slightly faster because it avoids recreating the greetingTimes array every time displayWelcomeMessage is called.